### ISSUE TITLE
Multiple vars bug in facet_grid in github version of ggplot

### ISSUE THREAD

**Issue by teunbrand** [tidyverse/ggplot2#3767]
When I installed the latest version of ggplot from github, `facet_grid()` started throwing errors when multiple variables are declared as the rows / columns. I can just work with the CRAN version, but thought it might be useful to report. Here is a reprex with an example from the documentation. 

Beware that this installs the github version of ggplot2 before blindly executing the reprex!

``` r
devtools::install_github("tidyverse/ggplot2", quiet = T, force = T)

library(ggplot2)

mg <- ggplot(mtcars, aes(x = mpg, y = wt)) + geom_point()
mg + facet_grid(vs + am ~ gear)
#> Error in gtable_add_grob(panel_table, strips$y$right, panel_pos_rows$t, : Not all inputs have either length 1 or same length same as 'grobs'

mg <- ggplot(mtcars, aes(x = mpg, y = wt)) + geom_point()
mg + facet_grid(vars(vs, am), vars(gear))
#> Error in gtable_add_grob(panel_table, strips$y$right, panel_pos_rows$t, : Not all inputs have either length 1 or same length same as 'grobs'
```

![](https://i.imgur.com/65I0ZUG.png)

<sup>Created on 2020-01-31 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

The following works without errors:

```
mg <- ggplot(mtcars, aes(x = mpg, y = wt)) + geom_point()
mg + facet_grid(vars(am), vars(gear))
```

<details>
  <summary>Session Info</summary>
<p>

```r
R version 3.6.2 (2019-12-12)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 16.04.6 LTS

Matrix products: default
BLAS:   /usr/lib/openblas-base/libblas.so.3
LAPACK: /usr/lib/libopenblasp-r0.2.18.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8     LC_MONETARY=en_US.UTF-8   
 [6] LC_MESSAGES=en_US.UTF-8    LC_PAPER=en_US.UTF-8       LC_NAME=C                  LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] ggplot2_3.3.0.9000

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.3        compiler_3.6.2    pillar_1.4.3      prettyunits_1.1.1 remotes_2.1.0     tools_3.6.2       testthat_2.3.1    digest_0.6.23    
 [9] pkgbuild_1.0.6    pkgload_1.0.2     evaluate_0.14     memoise_1.1.0     lifecycle_0.1.0   tibble_2.1.3      gtable_0.3.0      pkgconfig_2.0.3  
[17] rlang_0.4.4       reprex_0.3.0      cli_2.0.1         rstudioapi_0.10   curl_4.3          xfun_0.11         knitr_1.26        dplyr_0.8.3      
[25] withr_2.1.2       desc_1.2.0        fs_1.3.1          devtools_2.2.1    tidyselect_0.2.5  rprojroot_1.3-2   grid_3.6.2        glue_1.3.1       
[33] R6_2.4.1          processx_3.4.1    fansi_0.4.1       rmarkdown_2.0     sessioninfo_1.1.1 clipr_0.7.0       whisker_0.4       farver_2.0.3     
[41] purrr_0.3.3       callr_3.4.1       magrittr_1.5      htmltools_0.4.0   backports_1.1.5   scales_1.1.0      ps_1.3.0          ellipsis_0.3.0   
[49] usethis_1.5.1     assertthat_0.2.1  colorspace_1.4-1  labeling_0.3      munsell_0.5.0     crayon_1.3.4  
```

</p>
</details>

**Comment by yutannihilation** [comment]
Thanks, but this is already fixed in the RC version (`v3.3.0-rc` branch).

``` r
library(ggplot2)

mg <- ggplot(mtcars, aes(x = mpg, y = wt)) + geom_point()
mg + facet_grid(vs + am ~ gear)
```

![](https://i.imgur.com/XbuN3k8.png)

``` r

mg <- ggplot(mtcars, aes(x = mpg, y = wt)) + geom_point()
mg + facet_grid(vars(vs, am), vars(gear))
```

![](https://i.imgur.com/CrwBCPQ.png)

<sup>Created on 2020-01-31 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by clauswilke** [comment]
We ran into this at the tidydevday today. Thomas back-ported the fixes, so things should work again now.
